sklearn/datasets/_openml.py

```diff
@@ -782,15 +903,78 @@ def fetch_openml(
             "unusable. Warning: {}".format(data_description["warning"])
         )
 
-    return_sparse = False
-    if data_description["format"].lower() == "sparse_arff":
-        return_sparse = True
+    # TODO (1.3): remove "deprecated" from the valid parser
+    valid_parsers = ("auto", "pandas", "liac-arff", "deprecated")
+    if parser not in valid_parsers:
+        raise ValueError(
+            f"`parser` must be one of {', '.join(repr(p) for p in valid_parsers)}. Got"
+            f" {parser!r} instead."
+        )
+
+    if parser == "deprecated":
+        # TODO (1.3): remove this warning
+        parser = "liac-arff"
+        warn(
+            "The default value of `parser` will change from `'liac-arff'` to "
+            "`'auto'` in 1.3. You can set `parser='auto'` to silence this "
+            "warning. Therefore, an `ImportError` will be raised from 1.3 if "
+            "the dataset is dense and pandas is not installed.",
```

Comment:
I think users will be surprised when `parser='auto'` and their dataframes now have Int64 extension dtypes and the categorical are not strings anymore. I think it would be best to link to notes. If that can not be done:

```suggestion
            "the dataset is dense and pandas is not installed. Note that the pandas "
            "parser may return different data types. See the Notes Section in " 
            "fetch_openml's API doc for details.",
```